mmcv/utils/config.py

```diff
@@ -253,6 +253,32 @@ def fromfile(filename,
             import_modules_from_strings(**cfg_dict['custom_imports'])
         return Config(cfg_dict, cfg_text=cfg_text, filename=filename)
 
+    @staticmethod
+    def fromstring(cfg_str, file_format):
+        """Generate config from config str.
+
+        Args:
+            cfg_str (str): Config str.
+            file_format (str): Config file format corresponding to the
+               config str. Only py/yml/yaml/json type are supported now!
+
+        Returns:
+            obj:`Config`: Config obj.
+        """
+        if file_format not in ['.py', '.json', '.yaml', '.yml']:
+            raise IOError('Only py/yml/yaml/json type are supported now!')
+        with tempfile.NamedTemporaryFile('w', suffix=file_format) as temp_file:
+            temp_file.write(cfg_str)
+            temp_file.flush()
+            try:
+                cfg = Config.fromfile(temp_file.name)
+            except Exception as e:
```

Comment:
This exception handling is too wild. Other errors will also be catched and a misleading message will be printed.
You may use some trivial rules to check if this string represents python source codes. It may work for 90% cases though not perfect.
```python
if suffix != '.py':
    # check if users specify a wrong suffix for python
    if 'dict(' in cfg_str:
        warnings.warn('xxxx')
```

Reply:
I did want to trivial the exception rules but considering that the error type is difficult to define, soI did not do it.